projects/047 - Lai/template-sql/9-Alcohol-(skin-cohort).template.sql

```diff
@@ -0,0 +1,103 @@
+
+
+--> CODESET skin-cancer:1
+
+
+-- Set the start date
+DECLARE @StartDate datetime;
+DECLARE @EndDate datetime;
+SET @StartDate = '2011-01-01';
+SET @EndDate = '2022-06-01';
+
+--Just want the output, not the messages
+SET NOCOUNT ON;
+
+
+-- Create the skin cancer cohort=====================================================================================================================================
+IF OBJECT_ID('tempdb..#SkinCohort') IS NOT NULL DROP TABLE #SkinCohort;
+SELECT DISTINCT FK_Patient_Link_ID 
+INTO #SkinCohort
+FROM SharedCare.GP_Events
+WHERE (SuppliedCode IN (SELECT Code FROM #AllCodes WHERE (Concept = 'skin-cancer' AND [Version] = 1)))
+      AND EventDate >= @StartDate AND EventDate < @EndDate;
+
+
+-- Create a table with all patients for post COPI and within skin cohort=========================================================================================================================
+IF OBJECT_ID('tempdb..#PatientsToInclude') IS NOT NULL DROP TABLE #PatientsToInclude;
+SELECT FK_Patient_Link_ID INTO #PatientsToInclude
+FROM [SharedCare].[Patient_GP_History]
+WHERE FK_Patient_Link_ID IN (SELECT FK_Patient_Link_ID FROM #SkinCohort)
+GROUP BY FK_Patient_Link_ID
+HAVING MIN(StartDate) < '2022-06-01';
+
+IF OBJECT_ID('tempdb..#Patients') IS NOT NULL DROP TABLE #Patients;
+SELECT DISTINCT FK_Patient_Link_ID 
+INTO #Patients 
+FROM #PatientsToInclude;
+
+
+--> CODESET alcohol-non-drinker:1 alcohol-light-drinker:1 alcohol-moderate-drinker:1 alcohol-heavy-drinker:1 alcohol-weekly-intake:1
+
+
+IF OBJECT_ID('tempdb..#AllPatientAlcoholIntakeCodes') IS NOT NULL DROP TABLE #AllPatientAlcoholIntakeCodes;
+SELECT 
+	FK_Patient_Link_ID,
+	CAST(EventDate AS DATE) AS EventDate,
+	FK_Reference_Coding_ID,
+	FK_Reference_SnomedCT_ID,
+	[Value]
+INTO #AllPatientAlcoholIntakeCodes
+FROM [SharedCare].[GP_Events]
+WHERE FK_Patient_Link_ID IN (SELECT FK_Patient_Link_ID FROM #Patients)
+AND FK_Reference_SnomedCT_ID IN (
+	SELECT FK_Reference_SnomedCT_ID FROM #VersionedSnomedSets 
+	WHERE Concept IN (
+	'alcohol-non-drinker', 
+	'alcohol-light-drinker',
+	'alcohol-moderate-drinker',
+	'alcohol-heavy-drinker',
+	'alcohol-weekly-intake'
+	)
+	AND [Version]=1
+) AND CAST(EventDate AS DATE) < @EndDate
+UNION
+SELECT 
+	FK_Patient_Link_ID,
+	CAST(EventDate AS DATE) AS EventDate,
+	FK_Reference_Coding_ID,
+	FK_Reference_SnomedCT_ID,
+	[Value]
+FROM [SharedCare].[GP_Events]
+WHERE FK_Patient_Link_ID IN (SELECT FK_Patient_Link_ID FROM #Patients)
+AND FK_Reference_Coding_ID IN (
+	SELECT FK_Reference_Coding_ID FROM #VersionedCodeSets 
+	WHERE Concept IN (
+	'alcohol-non-drinker', 
+	'alcohol-light-drinker',
+	'alcohol-moderate-drinker',
+	'alcohol-heavy-drinker',
+	'alcohol-weekly-intake'
+	)
+	AND [Version]=1
+) AND CAST(EventDate AS DATE) < @EndDate;
+
+
+-- The final table========================================================================================================
+SELECT 
+	a.FK_Patient_Link_ID AS PatientId,
+	EventDate,
```

Comment:
"alcohol-weekly-intake" is not the same as the other categories in that you need to check the value (which is the number of units of alcohol per week) in order to put them into one of the categories. Check out the query-patient-alcohol-intake.sql file for an example of what I mean.

Reply:
Yes "alcohol-weekly-intake" was excluded from the script.